### PR TITLE
PLAT-864 Terraform SDK Upgrade

### PR DESCRIPTION
**Pull request by rjinskidepop**

### CODE REVIEW

File: catalog/data_source_catalog_env.go [hunk 1]
```diff
@@ -1,43 +1,45 @@
...
+			"format": {
 				Type:     schema.TypeString,
 				Required: true,
 			},
 		},
 	}
 }
 
-func dataSourceCatalogEnvDistinctRead(d *schema.ResourceData, meta interface{}) error {
+func dataSourceCatalogEnvDistinctRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
```
Comment (0x91) [inline catalog/data_source_catalog_env.go:35]: getServices should be refactored to use the context from the provider, currently it creates a TODO context

File: catalog/data_source_catalog_sandbox_kebab.go [hunk 2]
```diff
@@ -22,12 +24,13 @@ func dataSourceCatalogSandboxKebabDistinct() *schema.Resource {
 	}
 }
 
-func dataSourceCatalogSandboxKebabDistinctRead(d *schema.ResourceData, meta interface{}) error {
+func dataSourceCatalogSandboxKebabDistinctRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
```
Comment (0x91) [inline catalog/data_source_catalog_sandbox_kebab.go:27]: same comment as for dataSourceCatalogEnvDistinctRead

**Review by 0x91 — CHANGES_REQUESTED** [review]
we should pass through the new ctx into the catalog API client, otherwise we might hold up cancellations etc.

**Comment by rjinskidepop** [comment]
```
make testacc
==> Checking that code complies with gofmt requirements...
go test $(go list ./... |grep -v 'vendor') -cover -timeout 120m
?   	github.com/depop/terraform-provider-catalog	[no test files]
ok  	github.com/depop/terraform-provider-catalog/catalog	173.107s	coverage: 78.9% of statements
```

Time to run acceptance tests has increased but I assume this is the change to the testing strategy of the SDK. Everything looks correct and in-place 👍🏻

**Review by 0x91 — APPROVED** [review]

**Pull request merged**